.github/workflows/bench-primitives.yml

```diff
@@ -0,0 +1,106 @@
+name: rosenpass-ciphers - primitives - benchmark
+
+on:
+  pull_request:
+  push:
+
+env:
+  CARGO_TERM_COLOR: always
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  prim-benchmark:
+    strategy:
+      fail-fast: true
+      matrix:
+        system: ["x86_64-linux", "i686-linux"]
+
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Install nix
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@v27 # A popular action for installing Nix
+        with:
+          extra_nix_config: |
+            experimental-features = nix-command flakes
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+
+      # Set up environment
+
+      - name: 🛠️ Config Linux x64
+        run: echo "RUST_TARGET_FLAG=--target=x86_64-unknown-linux-gnu"  > $GITHUB_ENV
+        if: ${{ matrix.system == 'x86_64-linux' }}
```

Comment:
Friendly reminder: this logic is redundant to the Nix internal logic. If you just launch a devshell for the specific target you want, cargo is already correctly configured.

For example, when on my `x86_64-linux` I run `nix develop .\#packages.i686-linux.rosenpass`, then run `cargo build`, I will end up with a `ELF 32-bit LSB pie executable, Intel 80386`. For that I do not have to put any `--target` or similar, it _just works_ ™